### PR TITLE
task(settings,gql-api): Remove async requirements from relier

### PR DESCRIPTION
**Pull request by dschom**
## Because

- Async state for client Info and subscription info resulted in lots random async state
- This resulted in more useEffect calls and generally less clean code

## This pull request

- Makes the call to fetch client info when integration is first initialized
- Makes the call to fetch subscription on app start when integration
- Remove unneeded async status from many integration operations
- Adds support to fetch this data to graphql
- Fixes small bug where RP wasn't shown in confirm signup code header.

## Issue that this pull request solves

Closes: FXA-6836

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [ ] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Screenshots (Optional)

Please attach the screenshots of the changes made in case of change in user interface.

## Other information (Optional)

Note that in the ticket, it suggests we put this info in the configuration. After tinkering with that approach it had a couple drawbacks, mostly that it requires cross talk between the content-server and the auth-server. 

Since we are trying to push things to the graphql-api, the alternate approach suggested by @LZoog was taken and support was added to graphql-api instead. Apollo actually does a nice job of batching these startup calls into a single request, so the penalty does not appear to be that great. In the future we can even fetch this info via direct queries in the graphql-api if we prefer.

### CODE REVIEW

File: packages/fxa-settings/src/components/App/gql.ts [hunk 1]
```diff
@@ -53,3 +53,30 @@ export const GET_LOCAL_SIGNED_IN_STATUS = gql`
...
+ * This query fetches the current relying parties client information.
+ */
+export const GET_CLIENT_INFO = gql`
+  query GetClientInfo($input: String!) {
+    clientInfo(input: $input) {
+      clientId
+      imageUri
+      redirectUri
+      serviceName
+      trusted
```
Comment (dschom) [inline packages/fxa-settings/src/components/App/gql.ts:67]: :cry:

**Comment by vpomerleau** [comment]
I've only had a quick look at the code so far, but it's looking much much much cleaner! 💎 

However, when I run through the oauth reset password flow locally, I see this error after following the emailed reset password link (instead of seeing the `/complete_reset_password` page as expected):

image

**Comment by dschom** [comment]
> I've only had a quick look at the code so far, but it's looking much much much cleaner! 💎
> 
> However, when I run through the oauth reset password flow locally, I see this error after following the emailed reset password link (instead of seeing the `/complete_reset_password` page as expected):
> 
> image


Collateral damage from debugging the oauth signup test failures. I just pushed an update.

**Review by vpomerleau — APPROVED** [review]
Code looks good to me! 👍 Big improvement and much cleaner than before. Just have one question that isn't a blocker, and we've confirmed over Slack that the error I spotted before is unrelated to this PR (separate issue will be filed). :shipit:

File: packages/fxa-settings/src/components/App/index.tsx [hunk 2]
```diff
@@ -199,14 +208,21 @@ const AuthAndAccountSetupRoutes = ({
...
   // `serviceName` from this instead of calling integration.getServiceName() within pages
   const [serviceName, setServiceName] = useState<MozServices>();
+
   useEffect(() => {
+    // Don't initialize until integration is ready!
+    if (!integration) {
+      return;
+    }
+
     (async () => {
```
Comment (vpomerleau) [inline packages/fxa-settings/src/components/App/index.tsx:218]: Do we still need this to be async?

Comment (dschom) [reply packages/fxa-settings/src/components/App/index.tsx:218]: It doesn't. Nice catch.

File: packages/fxa-settings/src/models/integrations/oauth-integration.ts [hunk 3]
```diff
@@ -109,8 +108,7 @@ export class OAuthIntegrationData extends BaseIntegrationData {
   idTokenHint: string | undefined;
 
   @IsOptional()
-  @IsNumber()
-  @IsPositive()
+  @IsInt()
```
Comment (dschom) [inline packages/fxa-settings/src/models/integrations/oauth-integration.ts:111]: Note, that in our test cases maxAge will be provided as 0 sometimes.

File: packages/fxa-settings/src/components/App/index.tsx (hunk 2 above)
Comment (vpomerleau) [reply packages/fxa-settings/src/components/App/index.tsx]: Spotted another spot with aync `getServiceName` 🙈 
https://github.com/mozilla/fxa/blob/e4cb9723b5c8aeb138502220018a0c737cf58df4/packages/fxa-settings/src/pages/ResetPassword/index.tsx#L53

Comment (dschom) [reply packages/fxa-settings/src/components/App/index.tsx]: Me too :+1: . I caught a couple more and have pushed again.

**Pull request merged**